src/qibo/backends/abstract.py

```diff
@@ -122,6 +118,42 @@ def matrices(self):
             self._matrices = Matrices(self)
         return self._matrices
 
+    def circuit_class(self, accelerators=None, density_matrix=False):
+        """Returns class used to create circuit model.
+
+        Useful for hardware backends which use different circuit models.
+
+        Args:
+            accelerators (dict): Dictionary that maps device names to the number of
+                times each device will be used.
+                See :class:`qibo.core.distcircuit.DistributedCircuit` for more
+                details.
+            density_matrix (bool): If ``True`` it creates a circuit for density
+                matrix simulation. Default is ``False`` which corresponds to
+                state vector simulation.
+        """
+        if density_matrix:
+            if accelerators is not None:
+                raise_error(NotImplementedError, "Distributed circuits are not "
+                                                 "implemented for density "
+                                                 "matrices.")
+            from qibo.core.circuit import DensityMatrixCircuit
+            return DensityMatrixCircuit
+        elif accelerators is not None:
+            from qibo.core.distcircuit import DistributedCircuit
+            return DistributedCircuit
+        else:
+            from qibo.core.circuit import Circuit
+            return Circuit
```

Comment:
Maybe include some comment here talking saying that `qibo.core.circuit` is for simulation.

Reply:
I added some comments here and in `create_gates` and also for the setting hardware backend as default point from above. Let me know if you agree.